core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -408,6 +410,12 @@ class CivilizationInfo {
         else
             otherCiv.addNotification(meetString, NotificationIcon.Gold)
 
+        if (otherCiv.isCityState() && otherCiv.canGiveStat(Stat.Faith)){
+            otherCiv.addNotification(religionMeetString, NotificationIcon.Faith)
```

Comment:
I don't understand the need for the notifications to be split up. Why shouldn't they just be combined?

Reply:
1. the notification will be too long
2. one is a gold notification and one is a faith notification

Reply:
1. We have long notifications so that doesn't really matter imo
2. It could just be one stats notification with stats given as a token of goodwil? You can always add multiple icons if that is what you mean.

Reply:
I understand both views. This actually depends on how exactly Civ5 ticks for RCS, which I don't know... e.g. Gold is currently doubled for being the first major to meet a CS, Faith is not..
- Notification length is not an issue - it very much so but there's worse places already, so we need auto-wrapping ones anyway in some future. Simplicity trumps here.
- Displayed icons are vararg, you can have as many as you want. Doing so dynamically is a bit more than 10 chars code, though. Still, beauty would trump that consideration. _If_ doubling faith for a first-meet is OK this _could_ do it:
    ```kotlin
        val giftAmount = Stats(gold = 15f)
        if (cityStateType == CityStateType.Religious) giftAmount.faith = 4f
        var meetString = "[${civName}] has given us [${giftAmount}] as a token of goodwill for meeting us"
        if (diplomacy.filter { it.value.otherCiv().isMajorCiv() }.count() == 1) {
            giftAmount.timesInPlace(2f)
            meetString = "[${civName}] has given us [${giftAmount}] as we are the first major civ to meet them"
        }
        val icons = giftAmount.map { it.key.notificationIcon }.toTypedArray()
        if (cityStateLocation != null)
            otherCiv.addNotification(meetString, cityStateLocation, *icons)
        else
            otherCiv.addNotification(meetString, *icons)
        for ((key, value) in giftAmount)
            otherCiv.addStat(key, value.toInt())
    ```
... and yes, if you grok how and why that val icons works - congrats, level-up!
- `if (otherCiv.isCityState() && otherCiv.canGiveStat(Stat.Faith))` - well, at this point `otherCiv.isMajorCiv()` is guaranteed true...